synchronizer/l1_rollup_info_producer.go

```diff
@@ -0,0 +1,365 @@
+	statistics                           l1RollupInfoProducerStatistics
+}
+
+// l1DataRetrieverStatistics : create an instance of l1RollupInfoProducer
+func newL1DataRetriever(ctx context.Context, ethermans []EthermanInterface,
+	startingBlockNumber uint64, SyncChunkSize uint64,
+	outgoingChannel chan l1SyncMessage, renewLastBlockOnL1 bool) *l1RollupInfoProducer {
+	if cap(outgoingChannel) < len(ethermans) {
+		log.Warnf("l1RollupInfoProducer: outgoingChannel must have a capacity (%d) of at least equal to number of ether clients (%d)", cap(outgoingChannel), len(ethermans))
+	}
+	ctx, cancel := context.WithCancel(ctx)
+	ttlOfLastBlock := ttlOfLastBlockDefault
+	if !renewLastBlockOnL1 {
+		ttlOfLastBlock = ttlOfLastBlockDefault
+	}
+	result := l1RollupInfoProducer{
+		ctx:                                  ctx,
+		cancelCtx:                            cancel,
+		syncStatus:                           newSyncStatus(startingBlockNumber, SyncChunkSize),
+		workers:                              newWorkers(ctx, ethermans),
+		filterToSendOrdererResultsToConsumer: newFilterToSendOrdererResultsToConsumer(startingBlockNumber),
+		outgoingChannel:                      outgoingChannel,
+		statistics:                           newRollupInfoProducerStatistics(startingBlockNumber),
+		ttlOfLastBlockOnL1:                   ttlOfLastBlock,
+		status:                               producerIdle,
+	}
+	return &result
+}
+
+// TDOO: There is no min/max function in golang??
+func min[T constraints.Ordered](a, b T) T {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func max[T constraints.Ordered](a, b T) T {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+func (l *l1RollupInfoProducer) reset(startingBlockNumber uint64) {
+	log.Warnf("producer: Reset L1 sync process to blockNumber %d", startingBlockNumber)
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+	l.syncStatus.reset(startingBlockNumber)
+	l.statistics.reset(startingBlockNumber)
+	l.workers.stop()
+	// Empty pending rollupinfos
+	l.emptyChannel()
+	l.filterToSendOrdererResultsToConsumer.reset(startingBlockNumber)
+	l.status = producerIdle
+}
+
+func (l *l1RollupInfoProducer) emptyChannel() {
+	for len(l.outgoingChannel) > 0 {
+		<-l.outgoingChannel
+	}
+}
+
+// verify: test params and status without if not allowModify avoid doing connection or modification of objects
+func (l *l1RollupInfoProducer) verify() error {
+	return l.syncStatus.verify()
+}
+
+func (l *l1RollupInfoProducer) initialize() error {
+	err := l.verify()
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = l.workers.initialize()
+	if err != nil {
+		log.Fatal(err)
+	}
+	if l.syncStatus.needToRenewLastBlockOnL1() {
+		log.Infof("producer: Need a initial value for Last Block On L1, doing the request (maxRetries:%v, timeRequest:%v)",
+			maxRetriesForRequestnitialValueOfLastBlock, timeRequestInitialValueOfLastBlock)
+		//result := l.retrieveInitialValueOfLastBlock(maxRetriesForRequestnitialValueOfLastBlock, timeRequestInitialValueOfLastBlock)
```

Comment:
It's needed because in the first run need to ask for the las block on L1. 
- I renamed the function `needToRenewLastBlockOnL1` for `isSetLastBlockOnL1Value` that I think that make more sense